doc/cephfs/upgrading.rst

```diff
@@ -1,3 +1,51 @@
+Upgrading the MDS Cluster
+=========================
+
+Currently the MDS cluster does not have built-in versioning or file system
+flags to support seamless upgrades of the MDSs without potentially causing
+assertions or other faults due to incompatible messages or other functional
+differences. For this reason, it's necessary during any cluster upgrade to
+reduce the number of active MDS for a file system to one first so that two
+active MDS do not communicate with different versions.  Further, it's also
+necessary to take standbys offline as any new CompatSet flags will propogate
+via the MDSMap to all MDS and cause older MDS to suicide.
+
+The proper sequence for upgrading the MDS cluster is:
+
+1. Reduce the number of ranks to 1:
+
+::
+
+    ceph fs set <fs_name> max_mds 1
+
+2. Deactivate all non-zero ranks, from the highest rank to the lowest, while waiting for each MDS to finish stopping:
```

Comment:
@batrick I'm unsure to understand this
> Deactivate all non-zero ranks

I must be missing something but shouldn't `ceph fs set <fs_name> max_mds 1` end up with only zero ranks?

Reply:
This doc is for luminous clusters. It was updated for mimic+.

Reply:
@batrick there is still a BZ targeted downstream for 3.3z2 which means it's still valid for luminous

Reply:
https://docs.ceph.com/docs/luminous/cephfs/upgrading/

I think I understand your confusion: ranks are numbered 0 to (max_mds-1).